### PR TITLE
Issue 3757 - V5 Kanban Board

### LINKED ISSUE #3757: V5 Kanban Board MVP - Functionality only
### Description

In order to get a functional version of V5 up and running we must have a means to access the kanban board

### Goals

- [ ] Allow the kanban board to be accessed from V5
- [ ] Where possible maintain styling with V5
- [ ] If the page opens in a new tab, avoid users from accessing other elements of V4

### Related Resources
The design for this one is being addressed in #3726

### PR DESCRIPTION
**Pull request by Amantini1997**
This fixes #3757 

#### Description
Import Kanban board in v5 and re-style it

#### Test cases
- Create new Issue/Risk
- Edit Issue/Risk (double click on the card)
- Open model in viewer from kanban board card
- Open model in viewer from edit modal

### CODE REVIEW

**Comment by sanmont3drepo** [comment]
The clone button doesnt look good:

image

**Review by The-Daniel — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/routes/dashboard/projects/board/board.styles.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+			${ThumbnailPlaceholder} {
+				background-color: ${({ theme }) => theme.palette.tertiary.lightest};
+				color: transparent
+			}
+		}
+	}
+`;
+
+export const Container = styled.div`
```
[the remark this answers is not in the document]
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/dashboard/projects/board/board.styles.ts:77]: I see your point, but I think in this case it is fine. With the viewer, we are not importing the components, rather we are creating a common wrapper that re-styles all of them. here, we are not in the viewer though and the component is specifically re-imported with some props passed to it.

I would prefer to keep the styling next to tbh, but let's also see what's @sanmont3drepo 's opinion about this

[the remark this answers is not in the document]
Comment (The-Daniel) [reply frontend/src/v5/ui/routes/dashboard/projects/board/board.styles.ts:77]: https://github.com/3drepo/3drepo.io/pull/3797#discussion_r1047458255
This is what he said when I did the same

**Comment by sanmont3drepo** [comment]
> The clone button doesnt look good:
> 
> image

The  changes created additional scroillbars:

image

Maybe we could just change 'clone issue'  to clone

**Comment by The-Daniel** [comment]
Should the issues and risks buttons in container/federations lists take you to the relevant board now?
image
Also is this what the tasks tab was supposed to be, and do we need to remove that now? I don't see it in the figma mockup
image

**Comment by mosment3DRepo** [comment]
> Should the issues and risks buttons in container/federations lists take you to the relevant board now? image Also is this what the tasks tab was supposed to be, and do we need to remove that now? I don't see it in the figma mockup image

Agreed, but I think that should be a separate issue as it's not release limiting for this bit

**Comment by Amantini1997** [comment]
> Should the issues and risks buttons in container/federations lists take you to the relevant board now? image Also is this what the tasks tab was supposed to be, and do we need to remove that now? I don't see it in the figma mockup image

@mosment3DRepo Is the `Issues and risks` tab supposed to replace the `Tasks` one?

**Comment by The-Daniel** [comment]
On creating a new issue the scrollbar is in the wrong layer. This means you can scroll the header and footer out of view
image

**Comment by carmenfan** [comment]
> @mosment3DRepo Is the `Issues and risks` tab supposed to replace the `Tasks` one?

Tasks should be renamed to tickets and reserved for custom tickets in the future i think... I think we should minimise the linkage to the issue & risks kanban board so when we tear it out it isn't too intertwined.. would prefer it to be only accessible through the `X Risks` and `X Issues` button and the hamburger button only...

Ultimately it's @mosment3DRepo's call

**Comment by The-Daniel** [comment]
> On creating a new issue the scrollbar is in the wrong layer. This means you can scroll the header and footer out of view image

This issue still seems to exist

**Comment by george-hadfield** [comment]
### Move empty text to centre of board (vertically) and change styling

Change typography to H2 and colour to base.main

image

**Comment by george-hadfield** [comment]
### Make single selects disabled until Fed or Con is selected

Ensure disabled styling is applied

image

**Comment by george-hadfield** [comment]
### Change circle button to primary button with icon

If this is possible it would be great to implement

Build
image

Design
image

**Comment by george-hadfield** [comment]
### Update single select styling for Fed/Con selection

Build
image

Design
https://www.figma.com/file/3FjNEOdQbcHwzquVZxUqMl/3D-Repo?node-id=2386%3A15321&t=7zr6gSmpmEaLdPGr-1
image

**Comment by george-hadfield** [comment]
### Ensure border colour for issues/risk cards in list match the border colour for Feds and Containers
Correct border colour: rgb(224, 229, 240) (not sure which this is in palette)

Build
image

**Comment by george-hadfield** [comment]
### Add a few more pixels to the spacing around the column headers

Build
image

Design
image

**Comment by george-hadfield** [comment]
### Increase space between each column to 15px

Build
image

Design
image

**Comment by george-hadfield** [comment]
### View on modal icon disappears on hover
I would remove the icon completely and just rely on users clicking the issue to open the modal

Build
image

**Comment by george-hadfield** [comment]
### If a column has 0 issues, make the styling of '0 Issues' disabled

Build
image

**Comment by george-hadfield** [comment]
### Make the modal have a minimum height
Use the height of properties as the minimum

image

**Comment by george-hadfield** [comment]
### Reduce spacing between single select and empty comments box

image

**Comment by george-hadfield** [comment]
### Increase internal column spacing to 13px

Build
image


Design
image

**Comment by george-hadfield** [comment]
### Add droppable state to columns
 Use teal focus used on inputs

image

**Review by george-hadfield — CHANGES_REQUESTED** [review]
Comments left

**Comment by Amantini1997** [comment]
> ### Add droppable state to columns
> Use teal focus used on inputs
> 
> image

The library we use to create a kanban board (`react-trello`) claims to allow this kind of customisation, but unfortunately it doesn't seem to work. I looked for conversations about it on forum threads, but there is not much

**Comment by george-hadfield** [comment]
### Remove white border form new issue button

image

**Comment by george-hadfield** [comment]
### Ensure disable single selects use the correct v5 disabled styling

Build
image

Design (https://www.figma.com/file/3FjNEOdQbcHwzquVZxUqMl/3D-Repo?node-id=1597%3A14377&t=fJVAy2IEYbLsCSa4-1)
image

**Comment by george-hadfield** [comment]
### Reduce padding for titles in Fed and Con select menu
Change bottom padding from 17px to 8px as shown in the screenshot below. This should be applied to both Federations and Containers titles.

image

**Comment by george-hadfield** [comment]
### Remove background colour from Fed and Con selection menu
also remove top spacing (currently the white area above the blue background)

Build
image

Design
image

**Comment by george-hadfield** [comment]
### Add 5px top-padding to column titles
Currently 0 padding set on this element. Please replicate what's shown in the screenshot below

image

File: frontend/src/v5/ui/v4Adapter/dialogs/boardDialogs.overrides.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+				${ContentIssues}, ${ContentRisks} {
+					height: fit-content;
+					padding: 0 16px;
+					margin-bottom: 0 0 16px;
+					box-sizing: border-box;
+				}
+			}
+
+			${ResourcesList} {
+				/* margin-top: 0; */
```
Comment (The-Daniel) [inline frontend/src/v5/ui/v4Adapter/dialogs/boardDialogs.overrides.ts:140]: This can be deleted

**Review by The-Daniel — CHANGES_REQUESTED** [review]

**Comment by george-hadfield** [comment]
### Change padding-bottom to 15px on column headers

image

**Review by george-hadfield — CHANGES_REQUESTED** [review]
One final change

**Review by The-Daniel — DISMISSED** [review]

File: frontend/src/v5/ui/v4Adapter/components/conditionalV5OrViewerScrollArea.component.tsx [hunk 3]
```diff
@@ -17,12 +17,18 @@
 import { forwardRef } from 'react';
 import { useLocation } from 'react-router-dom';
 import { ScrollbarProps } from 'react-custom-scrollbars';
-import { ViewerScrollAreaBase } from './viewerScrollArea.styles';
+import { isV5 } from '@/v4/helpers/isV5';
+import { ConditionalV5OrViewerScrollAreaBase } from './conditionalV5OrViewerScrollArea.styles';
 
-export const ViewerScrollArea = forwardRef(({ children, ...props }: ScrollbarProps, ref: any) => {
+export const ConditionalV5OrViewerScrollArea = forwardRef(({ children, ...props }: ScrollbarProps, ref: any) => {
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/v4Adapter/components/conditionalV5OrViewerScrollArea.component.tsx:23]: I dont get why renaming this one :/

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

**Review by sanmont3drepo — APPROVED** [review]

**Review by sanmont3drepo — APPROVED** [review]

**Review by The-Daniel — APPROVED** [review]

**Review by george-hadfield — APPROVED** [review]

**Pull request merged**